src/extension.ts

```diff
@@ -194,21 +175,46 @@ function config(section: string, scope?: vscode.ConfigurationScope) {
 	return vscode.workspace.getConfiguration(section, scope);
 }
 
-function workspaceFolder(folder: string) {
-	return vscode.workspace.getWorkspaceFolder(vscode.Uri.parse(folder));
+function getWorkspaceFolder(folderName: string): vscode.WorkspaceFolder {
+	return vscode.workspace.getWorkspaceFolder(vscode.Uri.parse(folderName));
 }
 
-function folderNames(folders: readonly vscode.WorkspaceFolder[] = vscode.workspace.workspaceFolders): string[] {
-	if (!folders) {
-		return [];
+function getFolderName(folder: vscode.WorkspaceFolder): string {
+	let result = folder.uri.toString();
+	if (result.charAt(result.length - 1) !== '/') {
+		result = result + '/';
+	}
+	return result;
+}
+
+function sortedWorkspaceFolders() {
+	const workspaceFolders = vscode.workspace.workspaceFolders;
+	if (workspaceFolders) {
+		return vscode.workspace.workspaceFolders.map(f => getFolderName(f)).sort(
+			(a, b) => {
+				return a.length - b.length;
+			});
 	}
-	return folders.map(folder => {
-		let result = folder.uri.toString();
-		if (result.charAt(result.length - 1) !== '/') {
-			result = result + '/';
+	return [];
+}
+
+function folderNames(folders: readonly vscode.WorkspaceFolder[] = vscode.workspace.workspaceFolders): string[] {
```

Comment:
nitpick: I would rename this to `prunedFolderNames` since there is logic around pruning children, otherwise it might get used in a situation later where pruning is unwanted

Reply:
As an alternative to the test from below, I would move this helper and it's dependant helpers to the `utils` file, and add a unit test validating that `["a/", "a/b/"]` gets pruned to `["a/"]`, that is about all we can do.